cdap-api/src/main/java/io/cdap/cdap/api/security/credential/CredentialProvisionerProfile.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.security.credential;
+
+import com.google.gson.Gson;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Defines a profile for credential provisioning.
+ */
+public class CredentialProvisionerProfile implements Secured<CredentialProvisionerProfileMetadata> {
```

Comment:
this is a bit of an awkward name with 3 nouns in a row, and with Provisioner and Profile already having a different meaning in CDAP. What do you think about naming it something like CredentialProfile or CredentialProvider?